src/main/scala/sample/stream/GroupLogFile.scala

```diff
@@ -25,21 +25,25 @@ object GroupLogFile {
     // read lines from a log file
     val logFile = new File("src/main/resources/logfile.txt")
 
-    import akka.stream.io.Implicits._ // add file sources to Source object or use explicitly: SynchronousFileSource(f)
-    Source.synchronousFile(logFile).
+    FileIO.fromFile(logFile).
       // parse chunks of bytes into lines
       via(Framing.delimiter(ByteString(System.lineSeparator), maximumFrameLength = 512, allowTruncation = true)).
       map(_.utf8String).
+      map {
+        case line@LoglevelPattern(level) => (level, line)
+        case line@other => ("OTHER", line)
+      }.
       // group them by log level
-      groupBy {
-        case LoglevelPattern(level) => level
-        case other                  => "OTHER"
+      groupBy(5, _._1).
+      fold(("", List.empty[String])) {
+        case ((_, list), (level, line)) => (level, line :: list)
```

Comment:
I know, however see below comment: I'm not sure the FanInShape is the only way to solve it, so I'm using this to try out what "I would like the solution to look like". Do you propose we don't update the samples to 2.0 currently?